tests/unit/base/Plugin_unittest.cc

```diff
@@ -0,0 +1,202 @@
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Written (W) 2016 Sanuj Sharma
+ */
+
+#include <../tests/unit/base/MockBaseClass.h>
+#include <shogun/kernel/GaussianKernel.h>
+#include <gtest/gtest.h>
+
+using namespace shogun;
+
+TEST(Plugin, load_close)
+{
+    Library library = loadLibrary("./libplugin.so");
+    Manifest manifest = library.manifest();
+    MetaClass<MockBaseClass> mock_class = manifest.classByName<MockBaseClass>("mock_class");
+    MetaClass<MockBaseClass> another_mock_class = manifest.classByName<MockBaseClass>("another_mock_class");
+    Some<MockBaseClass> mock_class_obj = mock_class.instance();
+    Some<MockBaseClass> another_mock_class_obj = another_mock_class.instance();
+
+    std::string mock_class_name = mock_class_obj->get_name();
+    std::string another_mock_class_name = another_mock_class_obj->get_name();
+
+    EXPECT_EQ(mock_class_obj->mockMethod(), 0);
+    EXPECT_EQ(another_mock_class_obj->mockMethod(), 0);
+    EXPECT_EQ(mock_class_name, "MockClass");
+    EXPECT_EQ(another_mock_class_name, "AnotherMockClass");
```

Comment:
@lisitsyn If I add `EXPECT_EQ(mock_class, another_mock_class)`, test will pass although these two are not exactly the same. This is because `MetaClass` has `return true;` in its `operator==`. I think this can be fixed if we compare `this->instane()->get_name()` in equaltiy operator of `MetaClass`. But this works only if every class has `get_name()`.